internal/planner/planner.go

```diff
@@ -1360,6 +1360,10 @@ func (p *Planner) planObjectComprehension(oc *ast.ObjectComprehension, iter plan
 
 func (p *Planner) planComprehension(body ast.Body, closureIter planiter, target ir.Local, iter planiter) error {
 
+	// Variables that have been introduced in this comprehension have
+	// no effect on other parts of the policy, so they'll be dropped
+	// below.
+	p.vars.Dup()
```

Comment:
We should be able to call `p.vars.Push(map[ast.Var]ir.Local{})` here instead of this new function. The `Get()` function will check the top of the stack, if not found continue to the next element, and so on. Maybe I'm missing something about why this new function is needed.

Reply:
🤦 Yep. Of course. I wasn't seeing the forest for the trees. Replacing.